woocommerce/woocommerce-ios
# [Mobile Payments] Make optional updates work with the new UI

A continuation of #4974, this implements the same new update UI for optional updates, replacing the old `UpdateViewController`.

The new logic still doesn't handle failure properly, so I had to update a few tests to ignore that for now, but they'll come back in #5036.

The PR ended up a bit large, but a lot of that is removing the old UI.

https://user-images.githubusercontent.com/8739/134664793-55646195-dbd2-4c3f-aeb0-e7e0b62d2f0c.mp4

## To test

1. On Xcode, edit the WooCommerce scheme, and enable `-simulate-stripe-card-reader` in the Arguments tab
2. In `StripeCardReaderService.start(_:)`, after the SDK has been initialized (I add this right after `Terminal.shared.delegate = self`), add the following line: `Terminal.shared.simulatorConfiguration.availableReaderUpdate = .available`
3. Run the app
4. Go to Settings > In-Person Payments > Manage Card Reader and connect to a reader
5. After the reader connects, it should show that there's an update available
6. Tap Update Card Reader to start the update
7. Tapping the cancel button should cancel the update. Allowing the update to finish should clear the message saying that there's an update available.


Update release notes:
